server/server.js

```diff
@@ -1,162 +1,206 @@
-      if(keysToRemove.indexOf(key) == -1){
-        p[key] = value;
-      }
-    }
-  );
-  return p;
-}
-
-let obtainUrl = (req) => {
-  let url = req.url.substr(1)
-  if(typeof(req.headers['referer']) == 'undefined' || req.headers['referer'] == '' || true) {
-    return url
-  } else {
-    console.log("Having referer")
-    h = req.headers['referer'].replace('https://crossorig.in/http://', '').replace('https://crossorig.in/https://', '').replace('https://crossorig.in/', '')
-    h += h.endsWith("/") ? "" : "/"
-    return h + url;
-  }
-}
-
-app.use(bodyParser.raw({
-  inflate: true
-}));
-
-app.use(function(req, res, next) {
-  req.rawBody = '';
-  req.setEncoding('utf8');
-
-  req.on('data', function(chunk) {
-    req.rawBody += chunk;
-  });
-
-  req.on('end', function() {
-    next();
-  });
-});
-
-app.get(/(http|https)(:)\/\/(.*)/, function (req, res) {
-  var url = obtainUrl(req);
-  console.log("Request url GET: " + url)
-  console.log(req.url)
-  res.header('Access-Control-Allow-Origin', '*');
-  res.header('Access-Control-Request-Method', 'GET, PATCH, PUT, POST, OPTIONS, DELETE');
-  res.header('Doge', 'SUCH CORS');
+const express = require('express')
+const request = require('request')
+const bodyParser = require('body-parser')
+const app = express()
+
+const { removeKeys, obtainUrl } = require('./helpers.js')
+const { REGEX_HTTP, FORBIDDEN_CLIENT_HEADERS } = require('./constants.js')
+
+app.use(
+  bodyParser.raw({
+    inflate: true,
+  })
+)
+
+app.use((req, res, next) => {
+  req.rawBody = ''
+  req.setEncoding('utf8')
+
+  req.on('data', chunk => {
+    req.rawBody += chunk
+  })
+
+  req.on('end', () => {
+    next()
+  })
+})
+
+app.get(REGEX_HTTP, (req, res) => {
+  const url = obtainUrl(req)
+
+  res.header('Access-Control-Allow-Origin', '*')
+  res.header(
+    'Access-Control-Request-Method',
+    'GET, PATCH, PUT, POST, OPTIONS, DELETE'
+  )
+  res.header('Doge', 'SUCH CORS')
   res.header('Server', 'PENZOIL DRINKERS HOGO V1')
-  let headers = Object.assign(removeKeys(req.headers, FORBIDDEN_CLIENT_HEADERS), {});
-  k = request(url,
-  {
-    headers: headers
-  });
-  k.on('response', (r) => {
-    r.headers['server'] = 'JAY PENZOIL DRINKERS HOGO V1';
-    delete r.headers['set-cookie'];
-    delete r.headers['connection'];
-    delete r.headers['cf-ray'];
-    delete r.headers['cf-visitor'];
-    delete r.headers['host'];
-    delete r.headers['x-forwarded-for'];
-    console.log("Finished GET (" + r.statusCode + ") " + url)
-  }).on('error', function(e){ console.log('Ehm... something went wrong');console.log(e); res.end()}).pipe(res);
-});
-
-app.post(/(http|https)(:)\/\/(.*)/, function (req, res) {
-  var url = obtainUrl(req);
-  console.log("Request url POST: " + url)
+
+  const headers = Object.assign(
+    removeKeys(req.headers, FORBIDDEN_CLIENT_HEADERS),
+    {}
+  )
+
+  const k = request(url, { headers })
+
+  k
+    .on('response', r => {
+      r.headers['server'] = 'JAY PENZOIL DRINKERS HOGO V1'
```

Comment:
You need to change this to V2 and add some LEO

Reply:
love this CR 🎉